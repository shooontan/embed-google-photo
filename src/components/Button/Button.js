// @flow
import * as React from 'react';
import styled from 'styled-components';

type Props = {
  title?: string,
  onClick?: Function,
};

export const Button = (props: Props) => {
  const { title, onClick } = props;
  return <StyledButton onClick={onClick}>{title}</StyledButton>;
};

Button.defaultProps = {
  title: 'button',
  onClick: () => {},
};

const StyledButton = styled.button`
  display: inline-block;
  position: relative;
  padding: 0.75em 1.75em;
  margin: 10px;
  background: linear-gradient(to right, #ffa013, #ffc61a);
  border: none;
  border-radius: 50px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  outline: none;
  user-select: none;
  z-index: 2;

  &:after {
    content: '';
    position: absolute;
    z-index: -1;
    bottom: -6px;
    left: 5%;
    height: 110%;
    width: 90%;
    opacity: 0.8;
    border-radius: 50px;
    background: linear-gradient(to right, #ffa013, #ffc61a);
    filter: blur(6px);
    transition: all 0.2s;
  }

  &:hover:after {
    filter: blur(4px);
    width: 100%;
    bottom: -3px;
    left: 0;
  }

  &:hover:active::after {
    filter: blur(10px);
  }
`;

export default Button;
