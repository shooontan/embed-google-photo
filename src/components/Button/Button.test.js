// @flow
import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import { Button } from './Button';

Enzyme.configure({ adapter: new Adapter() });

test('Button', () => {
  let testRenderer = TestRenderer.create(<Button />);
  expect(testRenderer).toMatchSnapshot();

  testRenderer = TestRenderer.create(<Button title="Submit!!!" />);
  expect(testRenderer).toMatchSnapshot();

  const mock = jest.fn();
  testRenderer = shallow(<Button onClick={mock} />);
  testRenderer
    .dive()
    .find('button')
    .simulate('click');
  expect(mock).toHaveBeenCalled();
});
