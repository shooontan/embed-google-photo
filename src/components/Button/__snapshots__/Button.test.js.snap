// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  padding: 0.75em 1.75em;
  margin: 10px;
  background: linear-gradient(to right,#ffa013,#ffc61a);
  border: none;
  border-radius: 50px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 2;
}

.c0:after {
  content: '';
  position: absolute;
  z-index: -1;
  bottom: -6px;
  left: 5%;
  height: 110%;
  width: 90%;
  opacity: 0.8;
  border-radius: 50px;
  background: linear-gradient(to right,#ffa013,#ffc61a);
  -webkit-filter: blur(6px);
  filter: blur(6px);
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c0:hover:after {
  -webkit-filter: blur(4px);
  filter: blur(4px);
  width: 100%;
  bottom: -3px;
  left: 0;
}

.c0:hover:active::after {
  -webkit-filter: blur(10px);
  filter: blur(10px);
}

<button
  className="c0"
  onClick={[Function]}
>
  button
</button>
`;

exports[`Button 2`] = `
.c0 {
  display: inline-block;
  position: relative;
  padding: 0.75em 1.75em;
  margin: 10px;
  background: linear-gradient(to right,#ffa013,#ffc61a);
  border: none;
  border-radius: 50px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 2;
}

.c0:after {
  content: '';
  position: absolute;
  z-index: -1;
  bottom: -6px;
  left: 5%;
  height: 110%;
  width: 90%;
  opacity: 0.8;
  border-radius: 50px;
  background: linear-gradient(to right,#ffa013,#ffc61a);
  -webkit-filter: blur(6px);
  filter: blur(6px);
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c0:hover:after {
  -webkit-filter: blur(4px);
  filter: blur(4px);
  width: 100%;
  bottom: -3px;
  left: 0;
}

.c0:hover:active::after {
  -webkit-filter: blur(10px);
  filter: blur(10px);
}

<button
  className="c0"
  onClick={[Function]}
>
  Submit!!!
</button>
`;
